dev/PushNotifications/PushNotificationManager.cpp

```diff
@@ -423,49 +484,89 @@ namespace winrt::Microsoft::Windows::PushNotifications::implementation
 
     void PushNotificationManager::UnregisterAll()
     {
-        THROW_HR_IF(E_NOTIMPL, !::Microsoft::Windows::PushNotifications::Feature_PushNotifications::IsEnabled());
+        bool unregisterCompleted{ false };
+        {
+            auto lock{ m_lock.lock_exclusive() };
+            unregisterCompleted = (!m_singletonForegroundRegistration || !m_comActivatorRegistration);
+        }
 
-        try
+        if (!unregisterCompleted)
         {
             Unregister();
+        }
 
+        {
```

Comment:
Should this block be the very first thing we do in this function?

Reply:
If that is set at the beginning of the functions, Unregister will throw because m_registering will be true :(